### PR TITLE
Update azure node related methods in platform_nodes and azure_utils

### PR DESCRIPTION
**Pull request by sidhant-agrawal**
This PR includes: 
 - Add support for force VM stop
 - Handle the vms/nodes list in low-level modules similar to what we do for other platforms

Also, Fixes: #3736 

Signed-off-by: Sidhant Agrawal <sagrawal@redhat.com>

### CODE REVIEW

**Comment by sidhant-agrawal** [comment]
This PR verification can be done once PR #3637 is merged.

**Review by suchita-g — CHANGES_REQUESTED** [review]

File: ocs_ci/utility/azure_utils.py [hunk 1]
```diff
@@ -315,28 +316,45 @@ def get_disk_state(self, volume_name):
...
+        """
+        Stop and Start Azure vm instances
+
+        Args:
+            vm_names (list): Names of azure vm instances
+            force (bool): True for non-graceful VM shutdown, False for
+                graceful VM shutdown
+
+        """
+        self.stop_vm_instances(vm_names, force=True)
```
[the remark this answers is not in the document]
Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py:359]: ack, will fix this.

File: ocs_ci/ocs/platform_nodes.py [hunk 2]
```diff
@@ -1721,22 +1721,23 @@ def __init__(self):
         super(AZURENodes, self).__init__()
         self.azure = azure_utils.AZURE()
 
-    def stop_nodes(self, nodes, timeout=540, wait=True):
+    def stop_nodes(self, nodes, timeout=540, wait=True, force=True):
```
[the remark this answers is not in the document]
Comment (sidhant-agrawal) [reply ocs_ci/ocs/platform_nodes.py:1724]: @ebenahar In other platforms, we are using `force=True` by default.
Should we keep `force=True` by default in Azure or use default value False according to API doc?

File: ocs_ci/utility/azure_utils.py [hunk 3]
```diff
@@ -315,28 +316,45 @@ def get_disk_state(self, volume_name):
...
-        )
-        result.wait()
+        for vm_name in vm_names:
+            result = self.compute_client.virtual_machines.start(
+                self.cluster_resource_group, vm_name
+            )
+            result.wait()
 
-    def stop_vm_instance(self, vm_name):
+    def stop_vm_instances(self, vm_names, force=True):
```
Comment (ebenahar) [inline ocs_ci/utility/azure_utils.py:333]: as discussed, let's set it here to False, to be aligned with Azure API default

[the remark this answers is not in the document]
Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py]: updated

File: ocs_ci/utility/azure_utils.py [hunk 4]
```diff
@@ -315,28 +316,45 @@ def get_disk_state(self, volume_name):
...
-            self.cluster_resource_group, vm_name
-        )
-        result.wait()
+        for vm_name in vm_names:
+            result = self.compute_client.virtual_machines.power_off(
+                self.cluster_resource_group, vm_name, skip_shutdown=force
+            )
+            result.wait()
+
+    def restart_vm_instances_by_stop_and_start(self, vm_names, force=True):
```
[the remark this answers is not in the document]
Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py]: updated

File: ocs_ci/ocs/platform_nodes.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sidhant-agrawal) [reply ocs_ci/ocs/platform_nodes.py:1724]: as discussed offline, keeping `force=True` here

File: ocs_ci/ocs/platform_nodes.py [hunk 5]
```diff
@@ -1721,22 +1721,23 @@ def __init__(self):
...
         """
         Stop Azure vm instances
 
         Args:
             nodes (list): The OCS objects of the nodes
             wait (bool): True for waiting the instances to stop, False otherwise
             timeout (int): time in seconds to wait for node to reach 'not ready' state.
+            force (bool): True for force VM stop, False otherwise
```
[the remark this answers is not in the document]
Comment (sidhant-agrawal) [reply ocs_ci/ocs/platform_nodes.py:1732]: as discussed offline, keeping `force=True` in platform_nodes.py

File: ocs_ci/utility/azure_utils.py (hunk 3 above)
Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py]: updated azure_utils.py

File: ocs_ci/ocs/platform_nodes.py [hunk 6]
```diff
@@ -1804,7 +1804,9 @@ def restart_nodes(self, nodes, timeout=900, wait=True):
                 node_names=node_names, status=constants.NODE_READY, timeout=timeout
             )
 
-    def restart_nodes_by_stop_and_start(self, nodes, timeout=540, wait=True):
+    def restart_nodes_by_stop_and_start(
+        self, nodes, timeout=540, wait=True, force=True
```
[the remark this answers is not in the document]
Comment (sidhant-agrawal) [reply ocs_ci/ocs/platform_nodes.py:1808]: as discussed offline, keeping `force=True` here

**Review by ebenahar — DISMISSED** [review]

**Review by suchita-g — DISMISSED** [review]
LGTM

**Comment by sidhant-agrawal** [comment]
Do not merge because it needs a small change after PR #3637 is merged.

**Review by suchita-g — APPROVED** [review]
LGTM

File: ocs_ci/utility/azure_utils.py [hunk 7]
```diff
@@ -297,18 +297,19 @@ def detach_volume(self, volume, node, timeout=120):
...
         """
-        result = self.compute_client.virtual_machines.restart(
-            self.cluster_resource_group, vm_name
-        )
-        result.wait()
+        for vm_name in vm_names:
+            result = self.compute_client.virtual_machines.restart(
+                self.cluster_resource_group, vm_name
+            )
+            result.wait()
```
[the remark this answers is not in the document]
Comment (sidhant-agrawal) [reply ocs_ci/utility/azure_utils.py:312]: @mbukatov 
`restart_vm_instances()` will be called from `restart_nodes()` in platform_nodes.py.
Nodes status checks will be taken care of in that function.

**Review by ebenahar — APPROVED** [review]

**Pull request merged**